apps/backend/src/mutations/InviteMutations.ts

```diff
@@ -104,77 +112,81 @@ export default class InviteMutations {
-      roleIds
-    );
-    if (isRoleClaimAuthorized === false) {
-      return rejection(
-        'User is not authorized to update invites to this user type',
-        { userId: agent?.id, roleIds }
-      );
-    }
-
-    const isCoProposerClaimAuthorized =
-      coProposerProposalPk === undefined ||
-      (await this.proposalAuth.isMemberOfProposal(agent, coProposerProposalPk));
-
-    if (isCoProposerClaimAuthorized === false) {
+  @Authorized()
+  @EventBus(Event.EMAIL_INVITES)
+  public async setCoProposerInvites(
+    user: UserWithRole | null,
+    args: SetCoProposerInvitesInput
+  ): Promise<Invite[] | Rejection> {
+    const { proposalPk, emails } = args;
+    const isAuthorizedToCreateInvites =
+      await this.coProposerClaimAuth.hasCreateRights(user, proposalPk);
+
+    if (!isAuthorizedToCreateInvites) {
       return rejection(
-        'User is not authorized to update invites to this proposal',
-        { userId: agent?.id, proposalPk: coProposerProposalPk }
+        'User is not authorized to create invites for this proposal'
       );
     }
 
-    const updatedInvite = await this.inviteDataSource.update(args);
-    if (args.claims?.roleIds) {
-      await this.setRoleClaims(updatedInvite.id, args.claims?.roleIds);
-    }
-    if (args.claims?.coProposerProposalPk) {
-      await this.setCoProposerClaims(
-        updatedInvite.id,
-        args.claims.coProposerProposalPk
-      );
-    }
-
-    return updatedInvite;
-  }
-
-  private async setRoleClaims(inviteId: number, roleIds: number[]) {
-    await this.roleClaimDataSource.deleteByInviteId(inviteId);
-
-    if (!roleIds) return;
+    const existingClaims =
+      await this.coProposerClaimDataSource.findByProposalPk(proposalPk);
+    const existingInvites = (await Promise.all(
+      existingClaims.map((claim) =>
+        this.inviteDataSource.findById(claim.inviteId)
+      )
+    )) as Invite[];
+    const existingEmails = existingInvites.map((invite) => invite.email);
+
+    const deletedEmails = existingEmails.filter(
+      (email) => !emails.includes(email)
+    );
+    const newEmails = emails.filter((email) => !existingEmails.includes(email));
 
-    if (roleIds.length > 0) {
-      await Promise.all(
-        roleIds.map((roleId) =>
-          this.roleClaimDataSource.create(inviteId, roleId)
-        )
-      );
-    }
-  }
+    const deletedInvites = existingInvites.filter((invite) =>
+      deletedEmails.includes(invite.email)
+    );
 
-  private async setCoProposerClaims(invCodeId: number, proposalPk: number) {
-    if (!proposalPk) return;
+    await Promise.all(
+      deletedInvites.map((invite) => this.inviteDataSource.delete(invite.id))
+    );
+    const newInvites = await Promise.all(
+      newEmails.map(async (email) =>
+        this.inviteDataSource.create({
+          createdByUserId: user!.id,
+          code: await this.generateInviteCode(),
+          email: email,
+          expiresAt: new Date(Date.now() + 365 * 24 * 60 * 60 * 1000),

```

Comment:
Should the expiry date offset be moved into a constant so it's easier to understand and change if need be?